plugins/org.jkiss.dbeaver.data.transfer/src/org/jkiss/dbeaver/tools/transfer/stream/exporter/DataExporterGeoJSON.java

```diff
@@ -17,123 +17,258 @@
+                    if (typeLower.contains(geoType)) {
+                        geometryIndex = i;
+                        break;
+                    }
                 }
+                if (geometryIndex >= 0) break;
             }
         }
-        // Fallback: detect geometry column by common column names if type detection failed
+
+        // Fallback: detect geometry column by configurable column names if type detection failed
         if (geometryIndex < 0) {
             for (int i = 0; i < columns.size(); i++) {
                 String colName = columns.get(i).getName();
                 if (colName != null) {
-                    String n = colName.trim().toLowerCase();
-                    if (n.equals("geometry") || n.equals("geom") || n.equals("the_geom") || n.equals("geography") || n.equals("geojson") || n.equals("shape") || n.equals("wkb_geometry")) {
-                        geometryIndex = i;
-                        break;
+                    String colNameLower = colName.trim().toLowerCase();
+                    for (String geoColumnName : geometryColumnNames) {
+                        if (colNameLower.equals(geoColumnName)) {
+                            geometryIndex = i;
+                            break;
+                        }
                     }
+                    if (geometryIndex >= 0) break;
                 }
             }
         }
-        out.write("{\"type\": \"FeatureCollection\", \"features\": [\n");
-        out.flush();
+
+        // Start GeoJSON FeatureCollection using JsonWriter in stream mode
+        jsonWriter.beginObject();
+        jsonWriter.name("type").value("FeatureCollection");
+        jsonWriter.name("features");
+        jsonWriter.beginArray();
+
+        // Flush to ensure header is written immediately (needed for tests)
+        jsonWriter.flush();
+        if (writer != null) {
+            writer.flush();
+        }
+
         firstFeature = true;
     }
 
     @Override
     public void exportRow(DBCSession session, DBCResultSet resultSet, Object[] row) throws IOException {
-        if (!firstFeature) {
-            out.write(",\n");
-        }
-        firstFeature = false;
-        out.write("    {\n");
-        out.write("      \"type\": \"Feature\",\n");
+        // Start Feature
+        jsonWriter.beginObject();
+        jsonWriter.name("type").value("Feature");
+
         // Geometry
-        out.write("      \"geometry\": ");
-        if (geometryIndex >= 0 && row[geometryIndex] != null) {
-            String geomStr = row[geometryIndex].toString();
-            if (geomStr.trim().startsWith("{") || geomStr.trim().startsWith("[")) {
-                out.write(geomStr);
+        jsonWriter.name("geometry");
+        Object geometryValue = (geometryIndex >= 0) ? row[geometryIndex] : null;
+        if (geometryValue instanceof DBGeometry) {
+            Object gisData = ((DBGeometry) geometryValue).getRawValue();
```

Comment:
We're using Java 21, not Java 8. Same for other occurrences.


```suggestion
        if (geometryValue instanceof DBGeometry geometry) {
            Object gisData = geometry.getRawValue();
```